edge-hub/src/Microsoft.Azure.Devices.Edge.Hub.CloudProxy/ClientTokenCloudConnection.cs

```diff
@@ -0,0 +1,331 @@
+// Copyright (c) Microsoft. All rights reserved.
+
+namespace Microsoft.Azure.Devices.Edge.Hub.CloudProxy
+{
+    using System;
+    using System.Threading.Tasks;
+    using Microsoft.Azure.Devices.Client;
+    using Microsoft.Azure.Devices.Edge.Hub.Core;
+    using Microsoft.Azure.Devices.Edge.Hub.Core.Cloud;
+    using Microsoft.Azure.Devices.Edge.Hub.Core.Identity;
+    using Microsoft.Azure.Devices.Edge.Util;
+    using Microsoft.Azure.Devices.Edge.Util.Concurrency;
+    using Microsoft.Extensions.Logging;
+    using static System.FormattableString;
+
+    /// <summary>
+    /// This class creates and manages cloud connections (CloudProxy instances)
+    /// </summary>
+    class ClientTokenCloudConnection : CloudConnection, IClientTokenCloudConnection
```

Comment:
Could this be named TokenCloudConnection?

Reply:
The reason I called it ClientTokenCloudConnection is because it is not used for if the EdgeHub token is used for authentication. It is used only if the client token is used for auth.